Tests/JSONFactoryTests.swift

```diff
@@ -0,0 +1,179 @@
+                
+                it("handles all cluster components") {
+                    expect(component).toNot(beNil())
+                    expect(component?.viewController()).to(beAKindOf(UITabBarController.self))
+                    expect(component?.children().count).to(equal(2))
+                }
+                
+                it("handles all components recursively") {
+                    let firstChildren = component!.children().first
+                    let secondChildren = component!.children().last
+                    
+                    expect(firstChildren?.viewController()).to(beAKindOf(StackViewController.self))
+                    expect(secondChildren?.viewController()).to(beAKindOf(StackViewController.self))
+                }
+                
+                it("does not handle not registered components") {
+                    let firstChildren = component!.children().first
+                    let secondChildren = component!.children().last
+                    
+                    expect(firstChildren?.children().count).to(equal(0))
+                    expect(secondChildren?.children().count).to(equal(1))
+                }
+            }
+            
+            context("when registering all available factories") {
+                let jsonFactory = JSONFactory()
+                factories.forEach { jsonFactory.register($0) }
+                let component = try! jsonFactory.produce(from: json)
+                
+                it("handles all components recursively") {
+                    let firstChildren = component!.children().first!
+                    let secondChildren = component!.children().last!
+                    
+                    expect(firstChildren.children().count).to(equal(1))
+                    expect(secondChildren.children().count).to(equal(3))
+                    expect(firstChildren.children()[0].viewController()).to(beAKindOf(UIViewController.self))
+                    expect(secondChildren.children()[1].viewController()).to(beAKindOf(StackViewController.self))
+                }
+            }
+        }
+        
+    }
+    
+}
+
+fileprivate class ButtonFactory: ComponentFactory {
+    fileprivate func typeName() -> String {
+        return "button"
+    }
+}
+
+fileprivate class LabelFactory: ComponentFactory {
+    fileprivate func typeName() -> String {
+        return "label"
+    }
+}
+
+fileprivate class TableViewFactory: ComponentFactory {
+    fileprivate func typeName() -> String {
+        return "table_view"
+    }
+}
+
+fileprivate extension Component {
+    func children() -> [Component] {
+        switch self {
+        case .view(builder: _, meta: _):
+            return []
+        case let .wrapper(builder: _, child: child, meta: _):
+            return [child]
+        case let .cluster(builder: _, children: children, meta: _):
+            return children
+        }
+    }
+}
```

Comment:
I think it shouldn't be needed (so shouldn't be exposed) but maybe I'm missing the point to use it?